examples_tests/61.BlitFilterTest/blit/formats/r11f_g11f_b10f.glsl

```diff
@@ -0,0 +1,78 @@
+#ifndef _NBL_GLSL_BLIT_R11FG11FB10F_INCLUDED_
+#define _NBL_GLSL_BLIT_R11FG11FB10F_INCLUDED_
+
+uint to11bitFloat(in float _f32)
+{
+	const uint f32 = floatBitsToUint(_f32);
+
+	if ((f32 & 0x80000000u) != 0u)
+		return 0u;
+
+	const uint f11MantissaMask = 0x3fu;
+	const uint f11ExpMask = 0x1fu << 6;
+
+	const int exp = int(((f32 >> 23) & 0xffu) - 127);
+	const uint mantissa = f32 & 0x7fffffu;
+
+	uint f11 = 0u;
+	if (exp == 128) // inf / NaN
+	{
+		f11 = f11ExpMask;
+		if (mantissa != 0u)
+			f11 |= (mantissa & f11MantissaMask);
+	}
+	else if (exp > 15) // overflow converts to infinity
+		f11 = f11ExpMask;
+	else if (exp > -15)
+	{
+		const int e = exp + 15;
+		const uint m = mantissa >> (23 - 6);
+		f11 = (e << 6) | m;
+	}
+
+	return f11;
+}
+
+uint to10bitFloat(in float _f32)
+{
+	const uint f32 = floatBitsToUint(_f32);
+
+	if ((f32 & 0x80000000u) != 0u) // negative numbers converts to 0 (represented by all zeroes in 10bit format)
+		return 0;
+
+	const uint f10MantissaMask = 0x1fu;
+	const uint f10ExpMask = 0x1fu << 5;
+
+	const int exp = int(((f32 >> 23) & 0xffu) - 127);
+	const uint mantissa = f32 & 0x7fffffu;
+
+	uint f10 = 0u;
+	if (exp == 128) // inf / NaN
+	{
+		f10 = f10ExpMask;
+		if (mantissa != 0u)
+			f10 |= (mantissa & f10MantissaMask);
+	}
+	else if (exp > 15) // overflow, converts to infinity
+		f10 = f10ExpMask;
```

Comment:
afaik small floats (16bit and less) do not have NaN or INF (who'd want to waste bits for that), they will just saturate to max value

Reply:
So I just remove the `converts to infinity` part of the comment?

Reply:
I'd really want you to first and foremost, use @Crisspl's common float encoders.

Reply:
Are you referring to [this](https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/glsl/format/encode.glsl) or something else?

Reply:
> and function would turn into 3

What does that mean?

Reply:
waiting for the refactor to find its way into `nbl/builtin/glsl/format` and `nbl/builtin/glsl/ieee754.glsl`

https://github.com/Devsh-Graphics-Programming/Nabla/blob/a16fec199ca775bfc41a0f8e03aff90ff37197bb/include/nbl/builtin/glsl/limits/numeric.glsl#L43

Reply:
> afaik small floats (16bit and less) do not have NaN or INF (who'd want to waste bits for that), they will just saturate to max value

[This page](https://www.khronos.org/opengl/wiki/Small_Float_Formats#R11F_G11F_B10F) makes me think it is not the case.
For example take 11bit float.
It has 5 bit exponent. Possible biased exponent values $\in [0, 31]$.
Bias is 15. If we assume 0 and 31 for NaN/inf, then possible unbiased exponent values $\in [-14, 15]$.
Min possible 11bit float value would be $2^{-14} \times (1 + 0) = 6.1035 \times 10^{-5}$ --which matches the min value given in that page.

Reply:
If in doubt, read the spec:
https://www.khronos.org/registry/DataFormat/specs/1.3/dataformat.1.3.html

Reply:
you're right they do have Inf and NaN
https://www.khronos.org/registry/DataFormat/specs/1.3/dataformat.1.3.html#fpformats

my sincere apologies

Reply:
Resolved, except

can't move [these guys](https://github.com/Devsh-Graphics-Programming/Nabla/blob/5374db1c4b601e8a11d057ea359cb45440f0fa42/include/nbl/builtin/glsl/limits/numeric.glsl#L77) because they depend on `limits/numeric.glsl` --it will cause a circular include.

Reply:
> If in doubt, read the spec: https://www.khronos.org/registry/DataFormat/specs/1.3/dataformat.1.3.html

you can clearly see that RG and B have different minimum and maximum values